src/app/services/marketplace.service.ts

```diff
@@ -6,40 +6,81 @@ import {
   MarketplacePkg,
   StoreInfo,
 } from '@start9labs/marketplace'
-import { combineLatest, filter, Observable, of, shareReplay } from 'rxjs'
-import { map, switchMap, tap } from 'rxjs/operators'
+import {
+  combineLatest,
+  distinctUntilChanged,
+  filter,
+  Observable,
+  share,
+  shareReplay,
+} from 'rxjs'
+import { first, map, switchMap } from 'rxjs/operators'
 
 import { HOSTS } from '../tokens/hosts'
 import { UrlService } from './url.service'
+import { Router } from '@angular/router'
 
 @Injectable()
 export class MarketplaceService extends AbstractMarketplaceService {
+  constructor(private router: Router) {
+    super()
```

Comment:
You can use `private readonly router = inject(Router)` instead of constructor so you don't have to have constructor and a super() call.